module/observations/helper.js

```diff
@@ -2819,36 +2823,66 @@ module.exports = class ObservationsHelper {
           let topLevelEntityId = bodyData[topLevelEntityType];
 
 
-          let roles = bodyData.role.split(",");
+          let tenantDetails = await userService.fetchPublicTenantDetails(tenantData.tenantId);
+          if (!tenantDetails.success || !tenantDetails?.data?.meta) {
+            throw { 
+              status: httpStatusCode.bad_request.status,
+              message: messageConstants.apiResponses.FAILED_TO_FETCH_TENANT_DETAILS,
+            };
+          }
 
-          let entityTypeArr = [];
-          for(let roleIndex=0;roleIndex<roles.length;roleIndex++){
+          let observableEntityKeys = tenantDetails.data.meta?.observableEntityKeys || [];
 
-          let rolesDocumentAPICall = await entityManagementService.userRoleExtension({
-            code: roles[roleIndex],
-            tenantId:solutionDocument[0].tenantId,
-            orgIds:{$in:['ALL',...solutionDocument[0].orgIds]}
-          },
-          ["entityTypes.entityType"])
-
-          if(rolesDocumentAPICall.success){
-            entityTypeArr.push(rolesDocumentAPICall.data[0].entityTypes[0].entityType)
-          }else{
+          if (!observableEntityKeys || observableEntityKeys.length === 0) {
```

Comment:
@borkarsaish65 we should not give error, if keys are not there don't validate

Reply:
done